app/services/api/v1/target_service.rb

```diff
@@ -9,6 +9,8 @@ def initialize(target_params, current_user, longitude, latitude)
       end
 
       def create
+        raise MaxTargetError if @current_user.targets.count >= Target::MAX_TARGETS
```

Comment:
What about adding this validation in the model instead of the controller? So that if later you needs to create a target from a service or something like that you don't have to validate it again there. wdyt?

Reply:
I think it's ok, we should only create targets by using the service object. If we need to create targets from another service, we can still use this service.